### PR TITLE
Deposit funds

### PR DESCRIPTION
**Pull request by ehsueh**
* Deployed to Rinkeby: 0xac1D964A7623028ec263925c733F62047Cd5CD72
* Should try and test all functions

### CODE REVIEW

**Comment by ehsueh** [comment]
Changes: 
* Added transferAllFunds
* Everything compiles (but not tested)
Latest deployment to Rinkeby: `0xfFcCfE8A29C9B0628a369467D3D761aCf61D72A0`

File: contracts/SoulFund.sol [hunk 1]
```diff
@@ -100,8 +143,6 @@ contract SoulFund is
         return super.supportsInterface(_interfaceId);
     }
 
-    function addBeneficiary() external override {}
-
     function whitelistNft(address _newNftAddress, uint256 _tokenId)
```
Comment (benjamin852) [inline contracts/SoulFund.sol:149]: function name unclear compared to actual functionality here

File: contracts/SoulFund.sol [hunk 2]
```diff
@@ -71,6 +81,39 @@ contract SoulFund is
         _unpause();
     }
 
+    function depositFund(uint256 soulFundId, address currency, uint256 amount) external override payable onlyRole(GRANTER_ROLE)  {
+
+        // require that currency exists or max has not been reached
+        require(currencyIndices[soulFundId][currency] > 0 && numCurrencies[soulFundId] < 5, "SoulFund.depositFund: max currency type reached.");
+
+        uint index = currencyIndices[soulFundId][currency];
```
Comment (benjamin852) [inline contracts/SoulFund.sol:92]: uint256?

File: contracts/SoulFund.sol [hunk 3]
```diff
@@ -71,6 +81,39 @@ contract SoulFund is
         _unpause();
     }
 
+    function depositFund(uint256 soulFundId, address currency, uint256 amount) external override payable onlyRole(GRANTER_ROLE)  {
+
+        // require that currency exists or max has not been reached
+        require(currencyIndices[soulFundId][currency] > 0 && numCurrencies[soulFundId] < 5, "SoulFund.depositFund: max currency type reached.");
```
Comment (jrmoore5) [inline contracts/SoulFund.sol]: Trying to find where this will be > 0 initially, >=?

**Pull request merged**